source/localizable/tutorial/service.md

```diff
@@ -216,97 +177,190 @@ export default Ember.Service.extend({
+This `div` will act as a place for the 3rd part map API to render the map to.
 
-module.exports = function(defaults) {
-  let app = new EmberApp(defaults, {
-    // Add options here
-  });
+```app/templates/components/location-map.hbs
+<div class="map-container"></div>
+```
 
-  // Use `app.import` to add additional libraries to the generated
-  // output files.
-  //
-  // If you need to use different assets in different
-  // environments, specify an object as the first parameter. That
-  // object's keys should be the environment name and the values
-  // should be the asset to use in that environment.
-  //
-  // If the library that you are including contains AMD or ES6
-  // modules that you would like to import into your application
-  // please specify an object with the list of modules as keys
-  // along with the exports of each module as its value.
-  app.import('vendor/gmaps.js');
+Next, update the component to append the map output to the `div` element we created.
 
-  return app.toTree();
-};
-```
+We provide the maps service into our component by initializing a property of our component, called `maps`.
+Services are commonly made available in components and other Ember objects by ["service injection"](../../applications/services/#toc_accessing-services).
+When you initialize a property with `Ember.inject.service()`,
+Ember tries to set that property with a service matching its name.
 
-### Accessing the Google Maps API
+With our `maps` service, our component will call the `getMapElement` function with the provided location.
+We append the map element we get back from the service by implementing `didInsertElement`,
+which is a [component lifecycle hook](../../components/the-component-lifecycle/#toc_integrating-with-third-party-libraries-with-code-didinsertelement-code).
+This function runs during the component render, after the component's markup gets inserted into the page.
 
-Now that we have the maps API available to the application, we can create our map utility.
-Utility files can be generated using Ember CLI.
+```app/components/location-map.js
+import Ember from 'ember';
 
-```shell
-ember g util google-maps
+export default Ember.Component.extend({
+  maps: Ember.inject.service(),
+
+  didInsertElement() {
+    this._super(...arguments);
+    let location = this.get('location');
+    let mapElement = this.get('maps').getMapElement(location);
+    this.$('.map-container').append(mapElement);
+  }
+});
 ```
 
-The CLI `generate util` command will create a utility file and a unit test.
-We'll delete the unit test since we don't want to test Google code.
-Our app needs a single function, `createMap`, which makes use of `google.maps.Map` to create our map element, `google.maps.Geocoder` to lookup the coordinates of our location, and `google.maps.Marker` to pin our map based on the resolved location.
+Finally open the template file for our `rental-listing` component and add the new `location-map` component.
```

Comment:
You may have noticed that `this.get('location')` refers to a property `location` we haven't defined. This property will be passed in to the component by its parent template.